censys/censys-definitions
# Add initial containerized protoc environment

This is an attempt at creating a reliable censys-definitions build environment so code generation is simple and consistent.

Before this change, building our protobufs required [some manual setup][rm] that was really only configured on a Mac. While a good setup, this tries to lock everything down within a Docker container so a Mac is not required and local machine configuration isn't required.

Note, this setup _only_ generates our Python code since that's what was immediately needed as part of [CEN-369]. The next step, if using this setup is desirable, would be getting our Golang dependencies inside of the container.

## Testing

1. Check out the branch and note that you're in a clean check out
2. `docker-compose up` to build the image and run `make python`
    - This will take a while
    - This could be sped up by us building and hosting the image in a registry like [gcr.io](https://cloud.google.com/container-registry/)
3. Note that this finishes successfully, and you're still in a clean check out
    - That means that generating the python code worked as before
4. Make a trivial change, like adding a new field to a `message` with some `.proto` file
5. Regenerate via `docker-compose up` (or just `docker-compose run task-runner`)
6. Note that you now have changes both to the `.proto` file and your generated python code

[rm]: https://github.com/censys/censys-definitions/blob/b402a5d3b090090f89028e511143a9ad2752fb2a/README.md#regenerating
[CEN-369]: https://censysio.atlassian.net/browse/CEN-369
